routes/api/contacts.js

```diff
@@ -1,25 +1,87 @@
-const express = require('express')
+const express = require("express");
 
-const router = express.Router()
+const {
+	listContacts,
+	getContactById,
+	removeContact,
+	addContact,
+	updateContact,
+} = require("../../models/contacts");
+const {
+	newContactValidator,
+	updateContactValidator,
+} = require("../../validators/contactsValidator");
 
-router.get('/', async (req, res, next) => {
-  res.json({ message: 'template message' })
-})
+const router = express.Router();
 
-router.get('/:contactId', async (req, res, next) => {
-  res.json({ message: 'template message' })
-})
+router.get("/", async (req, res, next) => {
+	const contacts = await listContacts();
+	res.send({
+		contacts,
+	});
+});
 
-router.post('/', async (req, res, next) => {
-  res.json({ message: 'template message' })
-})
+router.get("/:contactId", async (req, res, next) => {
+	const id = req.params.contactId;
+	const contact = await getContactById(id);
+	if (contact) {
+		res.send({
+			contact,
+		});
+	} else {
+		res.status(404).send({
+			message: "Not found",
+		});
+	}
+});
 
-router.delete('/:contactId', async (req, res, next) => {
-  res.json({ message: 'template message' })
-})
+router.post("/", async (req, res, next) => {
+	const body = req.body;
+	const { name, email, phone } = req.body;
 
-router.put('/:contactId', async (req, res, next) => {
-  res.json({ message: 'template message' })
-})
+	const isValid = newContactValidator.validate(body);
 
-module.exports = router
+	if (!name || !email || !phone) {
+		res.status(400).send({
+			message: "missing required name - field",
+		});
+	} else if (isValid.error) {
+		res.status(400).send({ message: isValid.error.message });
+	} else {
+		const contact = await addContact(body);
+		res.status(201).send({ contact });
+	}
+});
+
+router.delete("/:contactId", async (req, res, next) => {
+	const id = req.params.contactId;
+	const isDeleted = await removeContact(id);
+
+	if (isDeleted) {
+		res.send({ message: "contact deleted" });
+	} else {
+		res.status(404).send({ message: "Not found" });
+	}
+});
+
+router.put("/:contactId", async (req, res, next) => {
+	const id = req.params.contactId;
+	const body = req.body;
+	const { name, email, phone } = req.body;
+
+	if (!name && !email && !phone) {
+		res.status(400).send({ message: "missing fields" });
+	} else {
+		const updatedContact = await updateContact(id, body);
+		const isValid = updateContactValidator.validate(updatedContact);
```

Comment:
Toteż powinno być na wejściu do funkcji i jeżeli coś nie tak to przerywasz z błędem. :)

Reply:
A tu o ile dobrze rozumiem, ma zwalidować dane, jeśli będą poprawne to wykonać funkcję, jeśli nie to jej nie wykonywać i zwrócić błąd?